[Frio] Add collapsable panel groups for admin/site with anchors at headlines

JonnyTischbein opened the pull request:
Because scrolling through `/admin/site` can take some time I added ~~quick links for the headlines~~ collapsable panel-groups like in `/admin/users`

~~- subsubmenu only visible at the acording page~~
~~- anchor is scrolling while noticing the fixed header~~
- Added own `site.tpl` for frio

Feature I want to add:
- open panel from which the submit has been placed

It's more a prototype. What do you think ? What need to be improved ?

![screenshot_2018-11-21_17-28-08](https://user-images.githubusercontent.com/10757520/48854770-fd093200-edb2-11e8-9852-b42613ac3f54.png)
![screenshot_2018-11-21_17-28-27](https://user-images.githubusercontent.com/10757520/48854779-04304000-edb3-11e8-9fb2-bdddee4378f0.png)

AlfredSK commented:
I like the idea because that page is huge. But I think the submenus are inconsistent. We don't have them anywhere else in frio's GUI. Maybe we should split the site settings in 3 to 4 pages or use a drop down selection in the upper right corner of the right pane (like the context menu for posts and comments at the network page - where you can remove or star an item).

annando commented:
Although I do like this in some way, our current solution at other pages are inflatable areas (see the user settings). I think we should try to keep it consistent.

MrPetovan commented:
I like it, thank you for your work!

AlfredSK commented:
Ah! Yes. The accordion stuff.

JonnyTischbein commented:
I removed the subsubpages and added the collapsable panel-groups. I also edited the inital post in this PR.
I'm working on open the last panel-group after submitting

annando commented:
Do we really need an own template for frio? Or could this be done on the standard template as well?

JonnyTischbein commented:
As far as I know some themes e.g. vier doesn't use the collapsing panel-groups.

JonnyTischbein commented:
Do we have some guidelines about javascript and what to store?
In my last commit I used `localStorage` to store the id of the panel where the submit was placed to reopen it.

MrPetovan commented on view/theme/frio/templates/admin/site.tpl:
```diff
@@ -0,0 +1,364 @@
+		$(".collapse").on('show.bs.collapse', function(e) {
+			localStorage.setItem('activeTab', $(e.target).attr('id'));
+		});
+		var activeTab = localStorage.getItem('activeTab');
```
Instead of relying on `localStorage` that isn't always available, I'd suggest you pass the current `activeTab` as a hidden form input to generate the correct `/admin/site#activeTab` redirection URL that will trigger the accordion opening.

JonnyTischbein replied:
I thought about that too, but to show a panel we need a javascript call or set the html params in the template
- A javascript call would be possible but only with regex match to get the element from the url. (Or do you know a js function to read params from the URL like `$_GET` ?)
- the template could edit the javascript code and replace the element

both ways seem more like a workaround

MrPetovan replied:
We use Bootstrap forfrio, so I assume that the appropriate accordion panel will open according to the URL fragment (after the #). Please verify this before adding any custom code: http://getbootstrap.com/2.3.2/javascript.html#collapse

JonnyTischbein replied:
I already did. That's why I started to going into js. 
`/admin/site#admin-settings-general-collapse` for the first panel group isn't working.

According to the bootstrap doc you linked it needs an controll element with
>data-toggle="collapse" and a data-target

or the manual js call `$(".collapse").collapse()`

JonnyTischbein replied:
I found this solution:

    // Opening accordion based on URL
    var url = document.location.toString();
    if ( url.match('#') ) {
       $('#'+url.split('#')[1]).addClass('in');
    }



https://stackoverflow.com/a/24201355

What do you think? :)

MrPetovan commented:
There aren’t any official guidelines but in general, please only use `localStorageˋ for optional features as it may or may not be available on all browsers. It isn’t even a question of compatibility, some browsers allow to disable it.

JonnyTischbein commented:
> Nice, have you successfully tested this new version?

I tested
- saving value in every panel and getting redirect to previous panel
- open/close other panel after redirection/saving
- testing on mobile
- only tested Firefox on Linux, Firefox on Android and partly on Chrome on Android
- (the scroll offset in chrome on Android is a bit too far)

AlfredSK commented:
Is it possible the move the relocation warning out of the last accordion headline? Would be looking better. You could put the warning inside the collapsed part and use a bold font to make the warning obvious.

JonnyTischbein commented:
![screenshot_2018-11-24_22-48-04](https://user-images.githubusercontent.com/10757520/48973236-70bd7000-f03b-11e8-83a8-1abd1861d659.png)

@MrPetovan `L10n::t('Relocate')` is translated by:	`Recent relocation`

MrPetovan commented on view/theme/frio/templates/admin/site.tpl:
```diff
@@ -351,7 +351,9 @@
 		<div id="admin-settings-relocate-collapse" class="panel-collapse collapse" role="tabpanel" aria-labelledby="admin-settings-relocate">
-
+			<div class="alert alert-danger">
+				<b>WARNING!</b> {{$relocate_warning}}
```
Please move the `<b>WARNING! </b>` string to the translation label.

tobiasd replied:
You should not use the `<b>` tag to emphasise something, semantically the `<strong>` tag should be used.

JonnyTischbein replied:
I'll have to create a new PR for this, right ?

MrPetovan replied:
Yes, even though you can use the same branch.

tobiasd replied:
The sad thing is, that in most environments there is no visual difference between the emphasis `em` and strong emphasis `strong` to the italic `i` and bold `b` settings. Thanks for changing it!

MrPetovan commented on mod/admin.php:
```diff
@@ -1492,7 +1492,8 @@ function admin_page_site(App $a)
 		'$relay_title' => L10n::t('Message Relay'),
-		'$relocate' => L10n::t('Relocate - WARNING: advanced function. Could make this server unreachable.'),
+		'$relocate' => L10n::t('Relocate Instance'),
+		'$relocate_warning' => L10n::t('Advanced function. Could make this server unreachable.'),
```
Remember you have to reference this variable in all `site.tpl` files, including the base template and all the supported themes that override it.

JonnyTischbein replied:
Now I made everything bold. Just for the future: Is it possible to use formatting in `L10n` String ? Or should the template var be something like `'$var => '<b>' . L10n::t('A') . '</b> ' . L10n::t('B');' ?
I wanted to keep the formatting in the template.

MrPetovan replied:
It's okay, you can put HTML tags in the translation strings.
